### ISSUE TITLE
The bdboot.txt file occasionally gets corrupted leaving users without ability to connect

### ISSUE THREAD

**Issue by ASmith-** [RetroShare/RetroShare#688]
The bdboot.txt file occasionally gets corrupted leaving users without ability to connect

The Problem
During my use of the Retroshare communications application I have helped 6 different users who had a corrupted bdboot.txt file and unable to further connect to the Retroshare network as a direct result. Other peers have also helped Retroshare users with this very same issue. Today number 7 came to the Freenode IRC #retroshare chatroom and then to the Spotchat IRC #retroshare chatroom for help with this very same issue on their Windows Retroshare system. This same corruption issue is found in all OS, distros, I've delt with this on my Linux distros on 3 different occassions myself, 2x the Retroshare application didn't shut-down entirely and timely which corrupted the bdboot.txt file and 1x DHT hackers were able to corrupt my bdboot.txt file with IP Addresses that all pointed back to themselves.

I have encountered Retroshare users with this problem attempting to simply delete the old bdboot.txt in hopes a new one would be automatically rebuilt. I have in nearly all cases needed to provide them with a fresh unpacked copy of the bdboot.txt file in order for them to once again be able to use the Retroshare application.

Regardless if this corruption came locally by the system not shutting down properly or externally by hackers attacking their Retroshare system via DHT (which sadly happens to be the current Retroshare default network configuration) the user is largely dead in the water and often very frustrated when I am able to reach them via various means to lend them a hand and get them back online in Retroshare.

Possible Solutions
The bdboot.txt file propertys must be read/write to update that file so it cannot simply be made read only by the user and still function properly IMO. However the developers can immediately add a bdboot-copy.txt file to the PPA packages as well as the builders of Windows,Linux,Mac packages doing the same to provide a fresh archive well within reach of any Retroshare user with a corrupted bdboot.txt file to help bail them out in a timely manner.

The addition of a 'repair bdboot.txt file' button in the Options -- network tab would be another nice and useful addition which when selected would copy bdboot-copy.txt over the top of the existing bdboot.txt file effectively renewing it immediately upon the user rebooting the Retroshare application.

**Comment by ASmith-** [comment]
I added bdboot-copy.txt to my .deb RetroShare06 install build, other builders can easily do the same to provide a ready archive for a user of their Retroshare Build when and if the bdboot.txt file gets corrupted. 

Here are the easy steps I made, non-linux builders could adapt this for Windows, Mac-Osx.

/usr/RetroShare06/bdboot-copy.txt <-- Added File beside existing bdboot.txt

/DEBIAN
  md5sums
  1ffe29292056f97f4c303bf63ac1fc48  usr/share/RetroShare06/bdboot.txt
  1ffe29292056f97f4c303bf63ac1fc48  usr/share/RetroShare06/bdboot-copy.txt <--Added to reference the bdboot-copy.txt file addition.

The tiny archive copy of bdboot.txt renamed as bdboot-copy.txt will now be internally included with each of my Retroshare06 amd64.deb Xenial Linux install builds. This solution doesn't solve the issue, it simply provides the user and helper with a easier pathway to get that Retroshare user connecting once again to the Retroshare network.

**Comment by dreamflasher** [comment]
I just ran into this issue, retroshare says that bdboot.txt is missing or outdated. I hoped to find this file in this repo, but it's not there. Also reinstallation does not solve this issue.

**Comment by ASmith-** [comment]
Hi Marcel,
Sorry to hear of your running into this near frequent problem. I have repeatedly asked the RS Devs to merely add a generic bdroot.txt file to the .retroshare folder as a protected backup. Anyway try this backup I found in my home folder.
    On Monday, May 1, 2023 at 06:51:55 AM PDT, Marcel Ackermann ***@***.***> wrote:  
 
 


I just ran into this issue, retroshare says that bdroot.txt is missing or outdated. I hoped to find this file in this repo, but it's not there. Also reinstallation does not solve this issue.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>
  
107.184.58.218 61240
2.31.102.246 10633
73.34.174.78 51413
195.154.172.169 34642
212.129.19.188 23899
70.228.99.197 4433
180.56.157.56 22360
126.153.25.87 24874
212.129.19.188 49967
82.5.211.96 19802
124.72.145.50 1024
176.114.190.206 3242
31.148.12.153 11609
195.154.179.2 47904
106.158.55.74 19343
148.0.41.212 50287
81.84.227.164 6881
75.134.7.27 62348
174.127.247.240 51413
59.18.224.239 58017
77.103.19.152 51413
172.248.59.113 46201
76.71.137.149 13889
85.230.110.61 62348
77.128.215.94 26485
50.149.125.113 45674
222.2.12.71 10727
175.43.117.224 16001
115.132.190.237 27105
158.255.179.181 51413
220.71.17.99 6881
82.196.8.13 51413
95.121.242.217 12852
91.122.47.159 49001
88.123.52.205 62010
5.189.187.90 2222
37.201.5.220 5869
213.136.79.238 51448
188.143.31.62 21234
14.47.66.62 62348
198.48.181.66 51413
111.64.15.235 51413
223.217.58.208 31077
36.224.47.175 6881
84.123.192.139 51413
88.182.134.251 49787
212.129.19.188 35762
176.31.124.113 54067
67.204.201.108 51413
212.129.19.188 32190
89.248.171.137 6881
82.15.118.251 24096
92.171.27.146 5667
89.189.176.9 22400
46.105.247.240 51413
5.146.209.168 6881
212.129.19.188 53655
92.58.180.180 51413
65.92.225.154 51413
5.228.55.70 31404
177.133.67.206 43035
185.32.221.150 51758
203.81.152.159 49830
31.10.165.80 19437
183.88.35.245 56242
84.7.36.14 63542
125.132.185.158 51413
82.0.88.43 8621
47.206.1.32 51413
109.70.185.171 49001
91.121.136.132 7812
109.217.241.131 32508
176.26.50.168 6881
92.131.116.227 40332
213.148.160.88 53265
82.232.72.172 6881
60.50.40.185 11527
23.97.59.26 51205
195.154.181.225 49597
101.181.1.39 6881
208.115.151.81 40195
178.62.61.61 12345
85.214.132.24 51413
49.191.134.155 52935
82.181.158.146 47081
95.222.150.218 31504
69.65.90.37 6881
195.154.181.225 30332
94.4.170.9 61137
176.26.253.5 17492
95.79.148.31 54312
212.129.19.188 41332
88.123.224.50 65162
37.11.110.24 51413
194.176.114.54 51413
222.152.175.120 51413
195.154.172.169 22828
195.154.181.225 39873
195.154.179.2 53437
220.116.21.169 43611
151.224.223.36 43611
212.129.19.188 54745
88.189.118.152 51413
222.186.19.7 65008
91.179.214.146 6881
124.177.22.2 43611
13.78.126.249 28073
73.7.31.91 51413
80.161.148.228 6881
75.136.107.181 53574
223.72.82.216 4688
94.210.248.192 51413
84.242.69.237 8001
182.239.194.57 6881
96.224.247.85 25622
80.99.77.167 51413
111.251.139.61 6881
5.79.77.7 47700
188.84.156.181 6881
130.204.204.201 25517
212.129.19.188 54384
174.54.22.10 61195
166.62.206.57 56132
106.255.116.84 56713
72.194.72.119 6881
198.100.145.92 51413
216.113.26.139 35713
23.225.206.157 6881
173.238.40.180 1024
89.110.53.186 41210
52.230.16.145 36680
118.5.29.191 6881
95.85.156.19 12034
60.96.139.6 22983
119.49.85.136 51413
94.36.26.85 23788
128.79.86.171 1024
47.88.22.26 6881
31.10.116.199 51413
13.70.85.57 47553
5.67.162.38 56218
69.121.80.52 51413
191.217.60.126 12922
71.45.78.9 34674
46.217.153.233 6881
84.222.219.48 60928
71.17.145.91 44822
149.202.90.75 51413
195.154.181.225 44546
160.13.108.45 6881
72.208.56.82 51413
195.154.181.225 53121
212.129.19.188 25202
13.70.85.57 41551
195.154.179.2 51990
212.129.19.188 40140
87.92.107.38 51413
24.21.0.77 11104
84.120.85.57 9089
82.35.127.218 17890
114.76.116.81 6881
37.59.37.75 51413
24.139.3.241 21216
114.146.150.240 6881
45.32.37.194 6881
76.4.13.202 6881
176.240.34.204 59802
86.137.104.135 51413
106.104.115.72 6881
195.154.181.225 41403
195.178.9.9 51413
119.188.50.231 65002
195.154.179.2 46273
82.209.232.166 51413
188.143.62.55 29160
87.61.67.85 51413
78.228.114.202 19490
37.187.105.46 51413
82.137.44.170 43970
199.126.202.176 57274
199.21.136.228 61123
195.154.181.225 41887
82.54.129.47 36536
213.136.79.205 49877
109.167.207.157 11844
60.68.200.94 7411
13.70.85.57 23634
121.180.167.249 61835
73.11.26.40 22729
77.77.57.186 59376
101.100.130.240 50583
60.56.169.104 9998
77.186.70.63 50463
78.90.9.184 19542
71.229.108.86 60352
101.184.103.225 35044
13.78.126.249 28007
212.129.19.188 47014
195.154.179.2 23087
195.154.172.169 36307
109.61.111.151 51413
94.79.49.117 29799
88.207.243.238 55114
45.63.51.192 6881
195.154.179.2 45174
83.223.7.84 51413
139.59.140.245 51413
104.236.79.228 6881
85.244.94.212 26085
65.60.152.244 41958
92.112.18.77 26039
195.154.179.2 30455
192.241.211.161 51413
24.22.189.172 44459
74.101.103.35 60073
79.131.143.33 14515
186.204.51.218 6881
85.254.14.162 51374
195.154.172.169 45168
84.196.231.227 51413
147.96.81.145 51413
195.154.181.225 51884
87.229.109.220 51413
24.91.169.141 26948
188.163.38.83 5059
185.53.154.226 1024
95.154.22.60 6881
95.29.146.190 8236
195.42.102.24 51413
2.8.251.42 51413
136.24.67.181 42583
61.215.186.248 22638
151.238.120.175 22132
176.213.216.211 37775
79.22.153.189 54150
92.57.165.36 8621
73.163.38.7 51413
51.254.196.20 51413
60.50.40.185 9235
77.245.185.186 51413
184.173.78.211 51413
88.169.133.198 38789
92.2.109.165 51413
87.96.232.42 51413
47.208.232.236 47659
130.179.56.198 8621
61.79.224.143 51413
221.219.156.188 25368
188.36.3.57 25255
80.103.151.58 58828
85.73.112.254 26085
97.100.89.5 28096
77.229.226.107 6881
114.76.0.135 51413
212.129.19.188 39992
77.110.7.12 46658
139.47.9.62 25238
198.74.50.57 51413
109.191.34.244 64842
117.201.94.152 11968
109.171.62.99 28773
58.224.103.146 52413
92.27.136.182 6881
88.176.201.54 51413
100.16.13.251 57559
5.189.183.129 8712
94.253.91.164 62909
201.22.48.193 45995
69.64.57.97 51413
13.68.101.145 30390
151.236.55.177 51413
46.105.17.191 51413
13.68.101.145 49370
104.236.29.167 51413
195.154.179.2 23162
68.229.102.231 50871
194.176.114.54 28437
213.136.79.205 7154
194.199.26.8 7812
138.91.244.75 54482
179.43.148.67 62514
123.255.47.19 51413
73.75.241.69 51413
220.253.162.226 28479
176.31.182.119 51413
60.127.149.130 9525
13.77.90.181 33736
36.224.230.56 16881
148.74.224.108 6881
212.129.19.188 35478
82.215.173.5 40298
46.39.238.145 51413
46.201.243.242 12269
68.3.218.195 6881
177.140.144.254 40643
24.205.185.107 21340
13.70.153.125 52591
82.146.56.119 6881
89.212.164.140 54533
104.173.152.69 19304
78.60.4.214 6881
90.8.41.190 18938
46.4.90.202 51413
94.67.80.39 57844
138.91.244.75 44250
94.210.174.3 51413
105.225.65.189 50759
212.71.249.150 51413
82.211.130.137 27420
184.56.147.70 27653
213.58.164.167 12135
176.36.184.50 6881
52.29.58.236 6992
195.34.240.192 22184
83.156.222.231 34564
95.211.93.145 56095
52.230.16.145 36368
209.6.149.142 44822
118.163.92.217 6881
95.223.106.5 39731
94.137.100.15 51413
92.45.240.26 51622
109.87.142.68 49001
31.204.190.226 49001
50.37.82.98 25766
80.98.228.133 56949
37.48.89.180 41800
195.154.176.162 48242
62.163.167.224 63285
24.135.16.136 12353
139.216.130.130 43611
82.126.223.50 6881
220.132.127.60 19531
24.236.36.145 51413
195.154.172.169 33258
186.1.162.110 38137
80.140.143.43 37500
80.110.78.48 28733
82.127.42.235 39844
78.194.69.129 51413
86.123.158.68 55748
37.187.146.194 63425
78.45.211.131 51413
185.41.154.173 51413
108.24.141.4 51413
94.28.143.240 49001
79.115.255.239 26085
191.251.202.98 51413
183.135.173.156 8129
195.154.179.2 45728
46.228.6.35 9058
90.154.131.70 18359
108.7.185.121 38151
79.143.191.138 6997
77.241.197.145 6881
192.161.178.88 6881
79.143.191.138 7001
45.77.13.73 6881
86.38.169.81 6881
82.154.5.135 6881
91.220.220.5 8000
195.154.181.225 34370
23.96.202.54 45249
195.154.181.225 54246
82.211.130.137 29777
189.189.64.51 1029
84.188.242.105 40000
212.129.19.188 41544
195.154.181.225 36459
195.154.181.225 50868
144.132.233.83 11111
81.35.11.218 30887
82.233.213.100 48061
178.32.217.168 9893
23.96.202.54 41130
195.154.181.225 25189
68.149.129.166 15614
122.249.227.99 6881
37.204.57.124 45036
138.91.244.75 45521
195.154.181.225 29034
70.117.197.234 51413
79.143.191.138 7005
85.155.159.203 51413
45.79.73.133 6881
24.86.248.6 23443
188.243.226.235 51413
13.77.90.181 26448
195.154.181.225 23355
195.154.172.169 51061
112.210.254.219 51413
91.115.119.253 6881
37.187.99.32 6881
88.169.1.120 59258
157.7.128.226 51413
91.157.169.126 64647
109.235.66.210 51413
210.145.160.162 49074
90.37.50.117 11193
37.48.70.151 55645
192.169.6.4 10556
59.126.210.170 23419
173.239.230.114 31325
195.154.14.133 51414
90.190.144.179 59526
122.117.139.93 23201
153.201.83.69 6881
77.140.167.99 51413
2.235.245.54 51413
72.4.31.165 9953
94.192.217.84 49850
109.13.135.57 61880
86.216.51.47 33125
88.185.135.58 6881
37.187.97.134 51413
80.114.244.41 51413
104.35.73.100 51413
80.98.78.47 51414
31.10.139.182 17013
94.60.106.61 11865
221.121.199.179 9430
98.110.19.81 16572
162.104.4.10 51413
195.154.172.169 31339
37.187.118.205 51413
115.22.163.233 58568
195.154.172.169 29246
64.114.250.215 51413
202.217.169.253 27800
71.194.231.45 49446
5.138.59.178 11609
40.84.153.118 45752
195.154.172.169 38221
195.154.179.2 28198
52.230.16.145 31133
50.159.57.131 12697
195.154.179.2 32822
173.2.234.50 19492
1.64.195.120 51413
124.131.132.236 51413
24.118.191.131 1085
83.196.158.141 18814
37.135.71.47 13097
195.154.179.2 35798
105.101.12.12 12998
212.92.112.141 58836
77.56.64.19 2004
195.154.181.225 49133
5.164.177.77 49001
13.77.90.181 53422
104.208.36.117 54208
195.154.181.225 33407
13.77.90.181 36529
138.91.244.75 48949
85.25.218.207 6882
178.35.200.127 35645
213.58.164.167 22355
98.247.140.204 60708
185.32.221.150 50663
181.120.184.79 51412
93.11.175.188 24960
212.129.19.188 22323
212.129.19.188 41973
84.126.122.235 51413
121.28.254.253 48749
85.72.191.125 33025
104.208.36.117 49945
23.97.59.26 52588
46.191.226.4 44050
79.115.37.141 39763
71.84.135.79 52296
99.137.6.202 62265
112.185.147.43 63884
96.40.135.104 57911
176.104.213.124 16947
81.97.66.21 13233
213.183.56.154 10099
70.48.62.57 12692
71.57.127.61 51753
77.203.253.126 33334
213.135.121.103 51413
178.132.74.58 10085
217.103.158.153 6881
213.213.184.216 49790
203.219.131.108 3668
212.129.19.188 39835
84.193.137.2 51413
50.116.50.18 51413
181.64.5.110 37792
134.249.140.119 51413
220.143.8.190 16956
89.84.201.175 39471
45.3.30.123 45324
195.154.181.225 51396
195.154.181.225 49939
99.3.37.82 26246
40.113.8.24 1048
187.147.252.175 40611
176.31.248.203 6883
45.32.69.51 6666
